### PR TITLE
Correct headings in engines guide [skip ci]

### PR DESCRIPTION
**Pull request by p8**
### Summary
Secondary headings should use ### instead of ## like everywhere else.

Before this change:

image

After this change:

image

### CODE REVIEW

File: guides/source/engines.md [hunk 1]
```diff
@@ -1505,7 +1505,7 @@ becomes
 ActiveSupport.on_load(:active_record) { self.include_root_in_json = true } # self refers to ActiveRecord::Base here
 ```
 
-## Available Hooks
+### Available Hooks
```
Comment (vipulnsward) [inline guides/source/engines.md]: This is probably a different section and not under above one

File: guides/source/engines.md [hunk 2]
```diff
@@ -1541,7 +1541,7 @@ To hook into the initialization process of one of the following classes use the
 | `ActiveSupport::TestCase`            | `active_support_test_case`           |
 | `i18n`                               | `i18n`                               |
 
-## Configuration hooks
+#### Configuration hooks
```
Comment (vipulnsward) [inline guides/source/engines.md]: Same

File: guides/source/engines.md (hunk 1 above)
Comment (p8) [reply guides/source/engines.md]: I thought they describe all the hooks that can be called with `on_load`. The available hooks list all the frameworks.

File: guides/source/engines.md (hunk 2 above)
Comment (p8) [reply guides/source/engines.md]: I've rolled back this change.

File: guides/source/engines.md (hunk 1 above)
Comment (p8) [reply guides/source/engines.md]: Maybe that text should be changed as well. For example `action_cable` is not a hook, but an argument that can be passed to `on_load`

**Comment by p8** [comment]
@vipulnsward I've made more changes to the headings:

-  I've combined the load and configuration hooks under a single heading to align more with the description at the top ("Avoid loading Rails frameworks with Load and Configuration Hooks").
- '`on_load` Hooks' has been changed to 'Load Hooks'.
- Headings have been changed to describe the underlying content

**Comment by vipulnsward** [comment]
Thanks!

**Pull request merged**